File: SjoslagetBackend/Models/BookingPax.cs
using System;

namespace Accidis.Sjoslaget.WebService.Models
{
	public sealed class BookingPax
	{
		public Guid Id { get; set; }
		public Guid BookingCabinId { get; set; }
		public string Group { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public Gender Gender { get; set; }
		public DateOfBirth Dob { get; set; }
		public string Nationality { get; set; }
		public int Years { get; set; }

		public static BookingPax FromSource(BookingSource.Pax source, Guid bookingCabinId)
		{
			return new BookingPax
			{
				BookingCabinId = bookingCabinId,
				Group = source.Group,
				FirstName = source.FirstName,
				LastName = source.LastName,
				Gender = GenderConvert.FromString(source.Gender),
				Dob = new DateOfBirth(source.Dob),
				Nationality = source.Nationality,
				Years = source.Years
			};
		}
	}
}

File: SjoslagetFrontend/lib/model/booking_pax.dart
import 'package:quiver/strings.dart' as str show isBlank;

class BookingPax {
	final String group;
	final String firstName;
	final String lastName;
	final String gender;
	final String dob;
	final String nationality;
	final int years;

	BookingPax(this.group, this.firstName, this.lastName, this.gender, this.dob, this.nationality, this.years);

	bool get isEmpty => str.isBlank(group) && str.isBlank(firstName) && str.isBlank(lastName) && str.isBlank(dob);
}
